File: apps/auth-server/src/main/java/com/example/chat/auth/server/infrastructure/persistence/JpaRefreshTokenRepository.java
package com.example.chat.auth.server.infrastructure.persistence;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface JpaRefreshTokenRepository extends JpaRepository<RefreshTokenEntity, UUID> {
    Optional<RefreshTokenEntity> findByTokenValue(String tokenValue);

    void deleteByTokenValue(String tokenValue);

    void deleteByPrincipalId(UUID principalId);
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/infrastructure/persistence/JpaRefreshTokenRepository.kt
package com.example.chat.auth.server.infrastructure.persistence

import java.util.*
import org.springframework.data.jpa.repository.JpaRepository

interface JpaRefreshTokenRepository : JpaRepository<RefreshTokenEntity, UUID> {
    fun findByTokenValue(tokenValue: String): Optional<RefreshTokenEntity>

    fun deleteByTokenValue(tokenValue: String)

    fun deleteByPrincipalId(principalId: UUID)
}
